include/openenclave/bits/eeid.h

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file eeid.h
+ *
+ * This file defines the EEID structure.
+ *
+ */
+
+#ifndef _OE_BITS_EEID_H
+#define _OE_BITS_EEID_H
+
+#include "properties.h"
+
+OE_EXTERNC_BEGIN
+
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+
+// This is the public key corresponding to the private key OE_DEBUG_SIGN_KEY in
+// signkey.c/.h.
+static const uint8_t OE_DEBUG_PUBLIC_KEY[] = {
+    0xca, 0x9a, 0xd7, 0x33, 0x14, 0x48, 0x98, 0x0a, 0xa2, 0x88, 0x90,
+    0xce, 0x73, 0xe4, 0x33, 0x63, 0x83, 0x77, 0xf1, 0x79, 0xab, 0x44,
+    0x56, 0xb2, 0xfe, 0x23, 0x71, 0x93, 0x19, 0x3a, 0x8d, 0xa};
+
+/*
+**==============================================================================
+**
+** oe_eeid_t
+**
+**==============================================================================
+*/
+typedef struct oe_eeid_t_
+{
+    struct
+    {
+        uint32_t H[8];
+        uint32_t N[2];
+    } hash_state; /* internal state of the hash computation at the end of
+                           the enclave base image */
+    uint64_t signature_size; /* size of signature */
+    uint8_t signature[1808]; /* base-image signature and associated data (for
+                           SGX, the complete sigstruct of the base image) */
+    oe_enclave_size_settings_t
+        size_settings; /* heap, stack and thread configuration for this instance
+                        */
+    uint64_t vaddr; /* location of the added data pages in enclave memory; EEID
+                       follows immediately thereafter */
+    uint64_t entry_point; /* entry point of the image */
+    uint64_t data_size;   /* size of application EEID */
+    uint8_t data[];       /* actual application EEID */
+} oe_eeid_t;
```

Comment:
Have a general question on the back/forward compatibility. There is no type version in the struct here. If we modify the struct or serializer/deserializer, how are we supporting earlier and future versions? Will it be a breaking change?

Reply:
Good point, version number has been added!